src/apps/components/AppWebhooksDisplay/EventDeliveriesList.tsx

```diff
@@ -0,0 +1,39 @@
+import { AppWebhookDeliveriesQuery } from "@dashboard/graphql";
+import React from "react";
+
+import { EventDeliveryItem } from "./EventDeliveryItem";
+
+type App = NonNullable<AppWebhookDeliveriesQuery["app"]>;
+type Webhook = NonNullable<App["webhooks"]>[0];
+type WebhookEventDelivery = NonNullable<Webhook["eventDeliveries"]>["edges"][0];
+
+export type EventDelivery = WebhookEventDelivery;
+
+interface EventDeliveriesListProps {
+  eventDeliveries: EventDelivery[];
+}
+
+export const EventDeliveriesList: React.FC<EventDeliveriesListProps> = ({ eventDeliveries }) => (
+  <>
+    {eventDeliveries.map((ed, index) => {
+      const { createdAt, id } = ed.node;
+      const attempts = ed.node.attempts?.edges?.map(({ node }) => node) ?? [];
+      const attemptsCount = attempts.length;
+      const lastAttemptDate = attempts[attemptsCount - 1]?.createdAt;
+      const hasMore = index < eventDeliveries.length - 1;
+
+      return (
+        <EventDeliveryItem
+          key={id}
+          dataTestId={id}
+          createdAt={createdAt}
+          status={ed.node.status}
+          attemptsCount={attemptsCount}
+          attempts={attempts}
+          lastAttemptDate={lastAttemptDate}
+          hasMore={hasMore}
+        />
+      );
+    })}
+  </>
+);
```

Comment:
In cases like this you could create something like DeliveryItem class or (even function + interfcace) and EventDeliveryItem component could take a single argument of this type.

That class can have static method eg. `createFrom(deliveryEntry: DeliveryEntry)` - in that case you have elegant separation and you are able to simply call:

```ts
{eventDeliveries.map(entry => <EventDeliveryItem  deliveryItem={DeliveryItem.from(entry)}  />)}
```

and this component is not needed anymore.

Think about that in the future, not needed now.